background/adapter/deluge.js

```diff
@@ -0,0 +1,70 @@
+if (typeof torrentToWeb === 'undefined') {
+    var torrentToWeb = {};
+};
+
+if (typeof torrentToWeb.adapter === 'undefined') {
+    torrentToWeb.adapter = {};
+};
+
+torrentToWeb.adapter.deluge = function(baseUrl, username, password)
+{
+    var baseUrlObject = new URL(baseUrl);
+    baseUrlObject.pathname = '/json';
+
+    baseUrl = baseUrlObject.toString();
+
+    function getSessionID() {
+        var request = new XMLHttpRequest();
+            request.open( "POST", baseUrl, true);
+            
+            request.send(
+		    JSON.stringify({
+			'id': '1',
+			'method': 'auth.login',
+			'params': [password]
+
+		    })
+	    );
+
+        return;
+    }
+
+getSessionID();
```

Comment:
I'm not exactly sure, this would log the user in immediately when the browser starts, so the session ID may invalidate meanwhile. I'd recommend to only call this within the send method.